server/src/main/java/org/elasticsearch/action/support/master/AcknowledgedResponse.java

```diff
@@ -75,7 +76,16 @@ protected void writeAcknowledged(StreamOutput out) throws IOException {
         out.writeBoolean(acknowledged);
     }
 
-    protected void addAcknowledgedField(XContentBuilder builder) throws IOException {
+    @Override
+    public final XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
         builder.field(ACKNOWLEDGED.getPreferredName(), isAcknowledged());
+        addCustomFields(builder, params);
+        builder.endObject();
+        return builder;
+    }
+
+    protected void addCustomFields(XContentBuilder builder, Params params) throws IOException {
```

Comment:
Wondering if this could be abstract, so subclasses don't forget to implement it. Looks like many would have an empty implementation though, so just a thought.

Reply:
it could be, but the classes that have a non-empty impl for it are more of an exception, so maybe it's ok like this?